pkg/utils/kube/ctl.go

```diff
@@ -213,7 +213,9 @@ func (k *KubectlCmd) DeleteResource(ctx context.Context, config *rest.Config, gv
 		zeroGracePeriod := int64(0)
 		deleteOptions.GracePeriodSeconds = &zeroGracePeriod
 	}
-
+	if cascadeDelete {
+		propagationPolicy = metav1.DeletePropagationOrphan
+	}
```

Comment:
I think since the `propagationPolicy` is an either/or (so, either foreground, background or orphaned), me might want to reflect that in the signature of `DeleteResource` method as well. With the current change in this PR, setting `forceDelete` and `cascadeDelete` to true would result in a kind of undefined behavior. Or is my assumption wrong, and `GracePeriodSeconds` should/can also be set for `orphaned` propagation policy?

If my assumption is correct, my proposal is, instead of introducing a new bool flag to the list of parameters (`cascadeDelete`), we should replace `forceDelete bool` with `propagationPolicy metav1.DeletionPropagetion` as the parameter, and rewrite logic in this method, as well as its callers.

Reply:
@jannfis Thanks for the review. I agree with what you suggested. I have addressed the review comment.

Reply:
Thanks, @jannfis for the review. I agree with your suggestion and have updated the PR with the logic.